motivaa-backend/motivaa-app/src/test/java/com/motivaa/control/HabitFinderServiceTest.java

```diff
@@ -0,0 +1,90 @@
+package com.motivaa.control;
+
+
+import com.motivaa.TestUtils.HabitFactory;
+import com.motivaa.control.error_handling.exceptions.RepositoryException;
+import com.motivaa.control.repository.MotivaaRepository;
+import com.motivaa.control.utility.MessageBundle;
+import com.motivaa.entity.Habit;
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import static org.mockito.Mockito.when;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.ActiveProfiles;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+@SpringBootTest
+@ActiveProfiles("test")
+public class HabitFinderServiceTest {
+    @MockBean
+    private MotivaaRepository motivaaRepository;
+
+    @Autowired
+    private HabitFinderService habitFinderService;
+
+    @Nested
+    class Success_scenarios {
```

Comment:
I am missing a test where you have multiple habits for different users, and it will only return the relevant ones.
Habit1 --> Reni's habit1
Habit2 --> Reni's second habit
Habit3 --> Roland's habit

search for Roland's uuid would return only Habit3

Reply:
That filtering is done in the ESclient (which is one of my questions for tuesday if I should write unit tests for that too), here if the motivaaRepository gives back 3 habits then the code returns 3 habits as a response - I don't see yet how that test could be done here

Reply:
Even thought right now the filtering is done only on the repository side I would add this unit test here, so it is clear and verified that only those should be returned which are related to the uuid. Let's discuss.